HLTrigger/btau/plugins/HLTTauDxyFilter.cc

```diff
@@ -0,0 +1,97 @@
+/** \class HLTTauDxyFilter
+ *
+ *  This class is an HLTFilter implementing
+ *  a minimum/maximum dxy cut on the tau object
+ *
+ */
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "HLTrigger/HLTcore/interface/HLTFilter.h"
+#include "DataFormats/TauReco/interface/PFTau.h"
+#include "DataFormats/TauReco/interface/PFTauFwd.h"
+#include "DataFormats/TauReco/interface/PFTauTransverseImpactParameterAssociation.h"
+
+#include <vector>
+#include <string>
+
+class HLTTauDxyFilter : public HLTFilter {
+public:
+  explicit HLTTauDxyFilter(const edm::ParameterSet& config);
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  bool hltFilter(edm::Event& event,
+                 const edm::EventSetup& setup,
+                 trigger::TriggerFilterObjectWithRefs& filterproduct) const override;
+
+private:
+  const edm::InputTag m_TausInputTag;
+  const edm::EDGetTokenT<reco::PFTauCollection> m_TausToken;
+  const edm::EDGetTokenT<edm::AssociationVector<reco::PFTauRefProd, std::vector<reco::PFTauTransverseImpactParameterRef>>>
+      m_TausTIPToken;
+
+  const double m_MinDxy, m_MaxDxy;  // dxy cuts applied to each tau
+  const int m_MinTaus;              // min. number of taus required to pass the cuts
+  const int m_TriggerType;
+};
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DataFormats/Common/interface/RefToBase.h"
+#include "DataFormats/HLTReco/interface/TriggerFilterObjectWithRefs.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
```

Comment:
```suggestion
```
Please move these header files upstairs with the other ones. I'm not sure `RefToBase.h` is needed, I would try to remove it.